File: src/Product.cpp
#include "Product.hpp"
#include <iostream>
using namespace std;
Product::Product() : productId(0), type(""), quantity(0), description(""), price(0.0) {}

Product::Product(int id, const string& type, int quantity, const string& description, double price) : productId(id), type(type), quantity(quantity), description(description), price(price) {}

void Product::incQuantity(int amount) {
    if (amount > 0) {
        quantity += amount;
        cout << "Увеличено количество товара '" << type << "' на " << amount<< ". Количество: " << quantity << endl;
    }
}
bool Product::decQuantity(int amount) {
    if (amount > 0 && quantity >= amount) {
        quantity -= amount;
        cout << "Уменьшено количество товара '" << type << "' на " << amount << ". Количество: " << quantity << endl;
        return true;
    }
    cout << "Ошибка: недостаточно товара '" << type << "' на складе" << endl;
    return false;
}
string Product::getInfo() const {
    string s = "ID: " + to_string(productId) + ", наименование: " + type + ", количество: " + std::to_string(quantity) + ", цена: " + std::to_string(price) +" руб., описание: " + description;
    return s;
}

int Product::getProductId() const { 
    return productId; 
}
string Product::getType() const { 
    return type; 
}
int Product::getQuantity() const {
    return quantity; 
}
string Product::getDescription() const {
    return description; 
}
double Product::getPrice() const {
    return price; 
}

File: warehouse/Product.java
package com.warehouse;

public class Product {
    private int productId;
    private String type;
    private int quantity;
    private String description;
    private double price;

    public Product() {
        this.productId = 0;
        this.type = "";
        this.quantity = 0;
        this.description = "";
        this.price = 0.0;
    }
    
    public Product(int id, String type, int quantity, String description, double price) {
        this.productId = id;
        this.type = type;
        this.quantity = quantity;
        this.description = description;
        this.price = price;
    }
    
    public void addQuantity(int amount) {
        if (amount > 0) {
            quantity += amount;
            System.out.println("Увеличено количество товара '" + type + "' на " + amount);
        }
    }
    
    public boolean lowQuantity(int amount) {
        if (amount > 0 && quantity >= amount) {
            quantity -= amount;
            System.out.println("Уменьшено количество товара '" + type + "' на " + amount);
            return true;
        }
        System.out.println("Ошибка: недостаточно товара '" + type + "' на складе");
        return false;
    }
    
    public String getInfo() {
        return "ID: " + productId + ", Тип: " + type + ", Количество: " + quantity + ", Цена: " + price + " руб., Описание: " + description;
    }
    
    public int getProductId() {

         return productId; 
    }
    public String getType() {

         return type; 
    }
    public int getQuantity() {

         return quantity;
    }
    public String getDescription() { 

        return description; 
    }
    public double getPrice() {
        
        return price; 
    }
}